e2e/app/component/table.ts

```diff
@@ -99,35 +99,28 @@ export default class Table extends Container {
     return this.page.waitForXPath(cellXpath, { visible: true });
   }
 
-  async getCellByValue(rowValue: string, columnHeader: string): Promise<Cell | null> {
-    // Find the row which contains cell
-    let rowIndex = -1;
-    const rows = await this.getRows();
-    for (let i = 0; i < rows.length; i++) {
-      const tds = await rows[i].$x('./td');
-      for (let j = 0; j < tds.length; j++) {
-        const v = await getPropValue<string>(tds[j], 'innerText');
-        if (v === rowValue) {
-          rowIndex = i;
-          break;
-        }
-      }
+  async findCellByRowValue(searchColumnHeader, searchCellValue, resultColumnHeader): Promise<Cell | null> {
+    // Find the searchColumnHeader index
+    const allColumns = await this.getColumns();
+    const colValues = await Promise.all(allColumns.map((column) => getPropValue<string>(column, 'innerText')));
+    console.log({ colValues });
+    const columnIndex = colValues.findIndex((v) => v === searchColumnHeader);
+    const resultColumnIndex = colValues.findIndex((v) => v === resultColumnHeader);
+    if (columnIndex === -1 || resultColumnIndex === -1) {
+      return null;
     }
 
-    // Find the column with correct header
-    let columnIndex = -1;
-    const columns = await this.getColumns();
-    for (let i = 0; i < columns.length; i++) {
-      if ((await getPropValue<string>(columns[i], 'innerText')) === columnHeader) {
-        columnIndex = i;
-        break;
-      }
-    }
+    // Find the row index which contains searchCellValue
+    const allRows = await this.getRows();
+    const tds = await Promise.all(allRows.map((row) => row.$x('./td')));
```

Comment:
There's a lot of extraneous state variables introduced compared to the previous approach which is making this more challenging to read, in my opinion. I'm thinking it may be more readable to group these together, e.g.:


```
const searchColumnTdValues = await Promise.all(allRows.map(async (row) => {
  const td = await row.$x('./td');
  return getPropValue<string>(td[columnIndex], 'innerText');
}))
```